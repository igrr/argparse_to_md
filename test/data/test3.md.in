test3: Subparsers test

<!--argparse_to_md:test3:get_parser:subheading_level=2-->
<!--argparse_to_md_end-->
