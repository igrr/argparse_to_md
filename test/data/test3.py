import argparse

def get_parser() -> argparse.ArgumentParser:
    parser = argparse.ArgumentParser(prog='testprog')
    subparsers = parser.add_subparsers(dest='action', help='action to run')

    foo_action = subparsers.add_parser('foo', description='foo action')
    foo_action.add_argument('--opt1', help='option 1')
    foo_action.add_argument('--opt2', help='option 2')

    bar_action = subparsers.add_parser('bar', description='bar action')
    bar_action.add_argument('--opt3', help='option 3')
    bar_action.add_argument('--opt4', help='option 4')

    parser.add_argument('--opt5', help='option 5')

    return parser

if __name__ == '__main__':
    parser = get_parser()
    parser.parse_args()
