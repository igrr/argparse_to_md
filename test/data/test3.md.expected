test3: Subparsers test

<!--argparse_to_md:test3:get_parser:subheading_level=2-->
## Usage:
```
testprog [-h] [--opt5 OPT5] {foo,bar} ...
```

Positional arguments:
- {`foo`, `bar`}: action to run

Optional arguments:
- `--opt5 OPT5`: option 5

## Usage of `foo`:

```
testprog foo [-h] [--opt1 OPT1] [--opt2 OPT2]
```
foo action

Optional arguments of `foo`:
- `--opt1 OPT1`: option 1
- `--opt2 OPT2`: option 2

## Usage of `bar`:

```
testprog bar [-h] [--opt3 OPT3] [--opt4 OPT4]
```
bar action

Optional arguments of `bar`:
- `--opt3 OPT3`: option 3
- `--opt4 OPT4`: option 4
<!--argparse_to_md_end-->
